scripts/mbedtls_framework/test_driver.py

```diff
@@ -0,0 +1,219 @@
+        need to be prefixed.
+        """
+        identifiers = set()
+        for file in iter_code_files(self.dst_dir):
+            identifiers.update(run_ctags(file))
+
+        identifiers_with_prefixes = set()
+        for identifier in identifiers:
+            if any(identifier.startswith(prefix) for prefix in prefixes):
+                identifiers_with_prefixes.add(identifier)
+        return identifiers_with_prefixes
+
+    def prefix_identifiers(self, identifiers_to_prefix: Set[str]):
+        """
+        In all test driver files, prefix each identifier in `identifiers_to_prefix`
+        with the test driver prefix: <DRIVER>_ for uppercase identifiers,
+        and <driver>_ for lowercase ones.
+        """
+        for f in iter_code_files(self.dst_dir):
+            self.__prefix_identifiers_in_file(f, identifiers_to_prefix, self.driver)
+
+    @staticmethod
+    def __rewrite_inclusions_in_file(file: Path, headers: Set[str],
+                                     src_include_dir: str, driver: str,) -> None:
+        """
+        Rewrite `#include` directives in `file` that refer to `src_include_dir/...`
+        so that they instead refer to `driver/...`.
+
+        For example:
+            #include "mbedtls/private/aes.h"
+        becomes:
+            #include "libtestdriver1/private/aes.h"
+        """
+        include_line_re = re.compile(
+            fr'^\s*#\s*include\s*([<"])\s*{src_include_dir}/([^>"]+)\s*([>"])',
+            re.MULTILINE
+        )
+        text = file.read_text(encoding="utf-8")
+        changed = False
+
+        def repl(m: Match) -> str:
+            nonlocal changed
+            header = m.group(2)
+            if header in headers:
+                changed = True
+                return f'#include {m.group(1)}{driver}/{header}{m.group(3)}'
+            return m.group(0)
+
+        new_text = include_line_re.sub(repl, text)
+        if changed:
+            file.write_text(new_text, encoding="utf-8")
+
+    @staticmethod
+    def __prefix_identifiers_in_file(file: Path, identifiers: Set[str], \
+                                     prefix: str) -> None:
+        """
+        In `file`, prefix each identifier in `identifiers` with the uppercase
+        form of `prefix` if the identifier is uppercase, or with the lowercase
+        form of `prefix` otherwise.
+        """
+        c_identifier_re = re.compile(r"\b[A-Za-z_][A-Za-z0-9_]*\b")
+        text = file.read_text(encoding="utf-8")
+        prefix_uppercased = prefix.upper()
+        prefix_lowercased = prefix.lower()
+
+        def repl(m: Match) -> str:
+            identifier = m.group(0)
+            if identifier in identifiers:
+                if identifier[0].isupper():
+                    return f"{prefix_uppercased}_{identifier}"
+                else:
+                    return f"{prefix_lowercased}_{identifier}"
+            return identifier
+
+        new_text = c_identifier_re.sub(repl, text)
```

Comment:
You can make regexps do the work:

```suggestion
        lower_identifier_re = re.compile(r'\b(' + '|'.join(ident for ident in identifiers if ident[0].islower()) + r')\b')
        upper_identifier_re = re.compile(r'\b(' + '|'.join(ident for ident in identifiers if ident[0].isupper()) + r')\b')
        text = file.read_text(encoding="utf-8")
        prefix_uppercased = prefix.upper()
        prefix_lowercased = prefix.lower()
        new_text = lower_identifier_re.sub(prefix.lower() + r'_\1',
                                           upper_identifier_re.sub(prefix.upper() + r'_\1', text))
```